Create lwm2m-agent-1020-440-28-resolving-auto-observe-and-validation-configuration-issue-for-a-resource.md (#3187)
SoftwareAG/c8y-docs: content/change-logs/device-management/lwm2m-agent-1020-440-28-resolving-auto-observe-and-validation-configuration-issue-for-a-resource.md

---
date: ""

> title: Resolving auto observe and validation configuration issue for a resource

Reviewer: ```suggestion
title: Resolved auto-observe and validation configuration issue for certain resources
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
ticket: DM-4586
version: 1020.440.28
---

> Resources that were supposed to have auto observe in LWM2M device protocol configuration activated could not be properly set up. The auto observe fragment was either always set to false or completely missing from the object. As a result, automatic observation was not executed. The auto observe fragment is now correctly set in the object, resolving the issue and ensuring that automatic observation works as expected. In addition, we addressed an issue where the empty validation rules configuration is saved for the resource. This caused issues for validations of write operation values before sending it to a LWM2M device. This issue has been resolved to ensure proper validation behavior.

Reviewer: ```suggestion
Resources that were supposed to have auto-observe in the LWM2M device protocol configuration activated could not be properly set up. The auto-observe fragment was either always set to false or completely missing from the object. As a result, automatic observation was not executed. The auto-observe fragment is now correctly set in the object, resolving the issue and ensuring that automatic observation works as expected. In addition, we addressed an issue where the empty validation rules configuration is saved for the resource. This caused issues for validations of write operation values before sending it to a LWM2M device. This issue has been resolved to ensure proper validation behavior.
```

> Resources that were supposed to have auto observe in LWM2M device protocol configuration activated could not be properly set up. The auto observe fragment was either always set to false or completely missing from the object. As a result, automatic observation was not executed. The auto observe fragment is now correctly set in the object, resolving the issue and ensuring that automatic observation works as expected.    In addition, we addressed an issue where the empty validation rules configuration is saved for the resource. This caused issues for validations of write operation values before sending it to a LWM2M device. This issue has been resolved to ensure proper validation behavior.    # Backport    This will backport the following commits from `develop` to `release/y2025`:   - [fix: [DM-4586] [Fix] Resolving autoObserve and validation configuration issue for a resource](https://github.com/Cumulocity-IoT/c8y-lwm2m/pull/330)    <!--- Backport version: 9.5.1 -->    ### Questions ?  Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)    [DM-4586]: https://cumulocity.atlassian.net/browse/DM-4586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Reviewer 2: ```suggestion
Resources that were supposed to have auto observe in LWM2M device protocol configuration activated could not be properly set up. The auto observe fragment was either always set to false or completely missing from the object. As a result, automatic observation was not executed. The auto observe fragment is now correctly set in the object, resolving the issue and ensuring that automatic observation works as expected. In addition, we addressed an issue where the empty validation rules configuration is saved for the resource. This caused issues for validations of write operation values before sending it to a LWM2M device. This issue has been resolved to ensure proper validation behavior.
```
